src/Microsoft.Extensions.Http.Polly/PolicyHttpMessageHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Polly;
+using System;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Http
+{
+    public class PolicyHttpMessageHandler : DelegatingHandler
+    {
+        private readonly IAsyncPolicy<HttpResponseMessage> _policy;
+
+        public PolicyHttpMessageHandler(IAsyncPolicy<HttpResponseMessage> policy)
+        {
+            if (policy == null)
+            {
+                throw new ArgumentNullException(nameof(policy));
+            }
+
+            _policy = policy;
+        }
+
+        protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request == null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            return _policy.ExecuteAsync(() => base.SendAsync(request, cancellationToken));
```

Comment:
Polly supports handing the cancellation token to the policy itself as well so it can be used by its internals (at least that's my understanding), which is passed through into the lambda doing the actual work. Should the cancellation token be used that way here?

Reply:
Agree, eg: 

    return _policy.ExecuteAsync(token => base.SendAsync(request, token), cancellationToken);

An example benefit: if the user has configured a wait-and-retry policy, a `cancellationToken` passed through the policy in this manner also governs the wait phase of wait-and-retry, so that signalling the `cancellationToken` will cause the policy to abandon the wait immediately.

Reply:
Yep, this is just something I missed. Will fix in the next iteration.